cs-ssh/src/main/java/io/cloudslang/content/ssh/services/impl/SSHServiceImpl.java

```diff
@@ -150,22 +159,32 @@ public CommandResult runShellCommand(
                 currentTime = System.currentTimeMillis();
             }
             boolean timedOut = !channel.isClosed();
-
-            // save the response
             CommandResult result = new CommandResult();
-            result.setStandardOutput(((ByteArrayOutputStream) out).toString(characterSet));
-            result.setStandardError(((ByteArrayOutputStream) err).toString(characterSet));
-
-            channel.disconnect();
-            // The exit status is only available after the channel was closed (more exactly, just before the channel is closed).
-            result.setExitCode(channel.getExitStatus());
-
             if (timedOut) {
                 throw new TimeoutException(String.valueOf(result));
             }
+            byte[] tmp = new byte[1024];
+            while (true) {
+                while (in.available() > 0) {
```

Comment:
Should simplify this. Can be rewritten with one loop and the condition should be return of `read`